docs/operator-manual/applicationset/Generators-Git.md

```diff
@@ -441,3 +441,13 @@ stringData:
 ```
 
 After saving, please restart the ApplicationSet pod for the changes to take effect.
+
+## Repository caching
+
+Files or directories that are added to or removed from the Git repository are only picked up by the ApplicationSet controller
```

Comment:
This doesn't seem like the desired behavior. The AppSet controller should be monitoring for new SHAs on the target revision and pulling new info from repo-server when a new SHA appears.

Reply:
Should I file a bug report instead?